app/assets/javascripts/filter-table.js

```diff
@@ -9,15 +9,15 @@ var combineQueryRows = function() {
       var $nextRow = $(rows[i+1])
       var nextRowQuery = $nextRow.find("td").first().text()
       if (currentRowQuery.toLowerCase().trim() === nextRowQuery.toLowerCase().trim()) {
-        $nextRow.addClass('govuk-table__row--group')
-        $currentRow.addClass('govuk-table__row--group')
+        $currentRow.addClass('govuk-table__row--group-fst')
+        $nextRow.addClass('govuk-table__row--group-snd')
```

Comment:
I love order bugs!

Reply:
It's not quite that, it's that with a sequence of `govuk-table__row--group`s, all but the first was hidden.  But you can have two adjacent-but-different groups, and the css just merged them.